src/pixel_utils.mjs

```diff
@@ -0,0 +1,137 @@
+import { PIXEL_DELIMITER, ROOT_PREFIX } from './constants.mjs';
+
+/**
+ * @typedef {Object} SearchExperimentDefinition
+ * @property {string} [description]
+ * @property {Array<string|number|boolean>} [variants]
+ */
+
+/**
+ * @typedef {Object} ParsedExperiment
+ * @property {string} key
+ * @property {string|null} description
+ * @property {Array<string|number|boolean>} [enum]
+ * @property {string} [type]
+ */
+
+/**
+ * Parses experiments matching schemas/search_experiments_schema.json5, remapping them to a format compatible with ignoreParams.
+ * @param {Record<string, SearchExperimentDefinition>} searchExperiments The raw search experiments.
+ * @returns {Record<string, ParsedExperiment>} Parsed experiments keyed by name and alternate name.
+ */
+export function parseSearchExperiments(searchExperiments) {
+    /** @type {Record<string, ParsedExperiment>} */
+    const out = {};
+
+    for (const [name, def] of Object.entries(searchExperiments)) {
+        out[name] = parseExperimentDef(name, def);
+        const altName = `prebounce_${name}`;
+        out[altName] = parseExperimentDef(altName, def);
+    }
+
+    return out;
+}
+
+/**
+ * Normalizes a single search experiment definition into the schema param representation.
+ * @param {string} name The experiment key.
+ * @param {SearchExperimentDefinition} def The raw experiment definition from SERP.
+ * @returns {ParsedExperiment} Experiment in pixel schema params format.
+ */
+function parseExperimentDef(name, def) {
+    const experiment = {
+        key: name,
+        description: def.description ?? null,
+    };
+
+    if (Array.isArray(def.variants)) {
+        experiment.enum = def.variants;
+        if (def.variants.length > 0) {
+            experiment.type = typeof def.variants[0];
+        }
+    }
+
+    return experiment;
+}
+
+/**
+ * Returns a lookup of search experiments with their enabled status.
+ * @param {Record<string, { addSearchExperimentParams?: boolean }>} pixels Parsed pixels keyed by experiment.
+ * @returns {Record<string, boolean>} A pixel mapping indicating which experiments are enabled.
+ */
+export function getEnabledSearchExperiments(pixels) {
```

Comment:
Do we want to keep the entire map in memory? Probably we only care about pixels that have experiments enabled or disabled. Minor since we will move away from this in the future.